### PR TITLE
Turf fires from horizion

### PR DESCRIPTION
**Pull request by retlaw34**
## About The Pull Request
This is by Azarak from horizion
Basicly, makes flamethrowers actually useful and make 
Turf fires!
Can be created by reageants igniting, or by explosion
They slowly raise the tempature of a room as well
port of https://github.com/hrzntal/horizon/pull/300

## Why It's Good For The Game
image

FIRE, FIRE, FIRE
## Changelog
:cl: Azarak
add: Turfs can now be ignited and will slowly burn, igniting anyone on them
add: Hotspots have 30% chance to ignite a turf, making fiery explosions leave cool residual flame.
balance: Flamethrowers have been reworked to use chemicals instead and now leave fire and hotspots on the turfs. They require an open reagent container instead of plasma tank now.
balance: Firealarms now trigger from 80C upwards (from 200C) or -10C downwards (from -3C). They also trigger immediately when they're on a turf fire
/:cl:

### CODE REVIEW

**Comment by Kitsunemitsu** [comment]
I love this. 
The people love this.
All we need is some more interactions.
I am approving.

**Comment by Azarak** [comment]
Make sure to include https://github.com/hrzntal/horizon/pull/558

**Comment by Azarak** [comment]
Also, dont port `connect_loc` for this. Just use the existing Crossed() proc. `connect_loc` exists so Crossed() and Uncrossed() don't have to exist. Implementation of `connect_loc` should properly conert all uses of those procs.

**Comment by retlaw34** [comment]
Ah, thank you

**Review by BootlegBow — APPROVED** [review]
we needed this back in 2016

File: code/game/objects/effects/turf_fire.dm [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+		var/atom/AT = A
+		AT.fire_act(TURF_FIRE_TEMP_BASE + (TURF_FIRE_TEMP_INCREMENT_PER_POWER*fire_power), TURF_FIRE_VOLUME)
+	if(!magical)
+		if(prob(fire_power))
+			open_turf.burn_tile()
+		if(prob(6))
+			playsound(open_turf, 'sound/effects/comfyfire.ogg', 40, TRUE)
+		UpdateFireState()
+
+/obj/effect/abstract/turf_fire/proc/on_entered(datum/source, atom/movable/AM)
```
Comment (Azarak) [inline code/game/objects/effects/turf_fire.dm:121]:
```suggestion
/obj/effect/abstract/turf_fire/Crossed(atom/movable/AM)
```

File: code/game/objects/effects/turf_fire.dm [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+/obj/effect/abstract/turf_fire/inferno/magical
+	magical = TRUE
+
+/obj/effect/abstract/turf_fire/Initialize(mapload, power)
+	. = ..()
+	var/turf/open/open_turf = loc
+	if(open_turf.turf_fire)
+		return INITIALIZE_HINT_QDEL
+	RegisterSignal(src, COMSIG_MOVABLE_CROSSED, .proc/on_entered)
```
Comment (Azarak) [inline code/game/objects/effects/turf_fire.dm:55]:
```suggestion
```

**Review by Azarak — COMMENTED** [review]
Not a component and Crossed() is an atom behaviour, so may aswell use it. Otherwise you need to unregister the signal

**Comment by Kitsunemitsu** [comment]
Passed a 5-0 Vote with one (early) code review.
Merging!

**Pull request merged**